### DISCUSSION TITLE
Tell us about the features you want in v9

### DISCUSSION THREAD

**Discussion by dolphin-wood** [idiotWu/smooth-scrollbar#392] [General]
Hi, folks! It has been 4 years since the first release of smooth-scrollbar@8. Thanks for your support and help :). The code and the workflows are a bit outdated (we are still using typescript@3 and bower in this project!), so I think it's time to draft a major update 🚀 . 

The planned changes have been listed in [project v9](https://github.com/idiotWu/smooth-scrollbar/projects), but if there is something more you want to add, feel free to start a new discussion in the [ideas category](https://github.com/idiotWu/smooth-scrollbar/discussions/categories/ideas).

👉 [List of accepted feature requests](https://github.com/idiotWu/smooth-scrollbar/issues?q=label%3Aaccepted)
👉 [Project v9](https://github.com/users/idiotWu/projects/1)

**Comment by dellyn** [comment]
Zoom feature - helpful feature
Scrolling by dragging the scroll content - would be nice, but it's easy to implement using custom calculation and mouseMove event

**Reply by dolphin-wood** [reply]
> Scrolling by dragging the scroll content

Actually, I just create a pannable plugin in another project with custom events 😆 : [keyframe-refiner/src/app/scrollbar-plugins.ts#L64-L118](https://github.com/keyframe-refiner/keyframe-refiner/blob/5497fe451411c612a4b0da501528b0c7e65f7106/src/app/scrollbar-plugins.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L64-L118).

👉 [Sample Usage](https://github.com/keyframe-refiner/keyframe-refiner/blob/5497fe451411c612a4b0da501528b0c7e65f7106/src/app/components/Cropper.svelte?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L70-L79)

**Comment by maroshvarchola** [comment]
Hello, it would be really good to have middlebutton scroll support, or create whole listener and custom mmb image etc.

**Reply by dolphin-wood** [reply]
This can currently be implemented as a plugin. I'll consider adding it to official plugins (or maybe the core?) in v9 :)

**Reply by dolphin-wood** [reply]
Related issue: https://github.com/idiotWu/smooth-scrollbar/issues/261

**Comment by xdloper** [comment]
It would be great if there was an onScroll feature. I should be able to perform scroll operations as in jquery, thanks in advance ;)

**Reply by dolphin-wood** [reply]
You meant [.addListener()](https://github.com/idiotWu/smooth-scrollbar/blob/develop/docs/api.md#scrollbaraddlistener)?

**Comment by InukaAsith** [comment]
paralax please

**Comment by Dushyant1295** [comment]
The browsers are well optimized these days so it's time to cut the codes that support old browser  and minimize the  size

**Comment by Dushyant1295** [comment]
This is the oldest virtual smooth library lots of award-winning site are built using this such as https://www.chungiyoo.com/
still this library is not popular like others i.e. locomotive scroll
it would be nice if we have nice portfolio site for this repo also along with SEO strategy / tags
